### ISSUE TITLE
Tutorial on using union-types

### ISSUE THREAD

**Issue by jethrolarson** [paldepind/union-type#33]
I think there's a lack of info on how to actually apply union types to real problems in js. 

I'd like to see something like http://fsharpforfunandprofit.com/series/designing-with-types.html
offered somewhere. I understand if you want that outside the scope of this repo, but I thought I'd open an issue incase you want to make it a wiki or something.

I'd be interested in contributing here.

**Comment by ericgj** [comment]
I agree, great idea. Quite a few of the examples in [functional-frontend-architecture](https://github.com/paldepind/functional-frontend-architecture/tree/master/examples) use this library. Granted, they are pretty narrowly focused on modelling actions in the Elm architecture sense, and they don't demonstrate the new features of union-type.  But that may be a place to collect some examples.  

I have a few half-baked projects which use union-type outside of the Elm architecture, I can't say they are more than getting my feet wet, but they may be useful to look at.

**Comment by paldepind** [comment]
I agree with this as well. Just having such a library is one thing. But figuring out how best to use it is probably not easy unless one has experience in a language with algebraic data types. A tutorial would be a great idea.

**Comment by kwijibo** [comment]
+1 The FSharp blog was also what got me interested in this style of programming, and while I've managed to find this library very useful, my understanding of (functional) type-based design is inferred too much from current capabilities of `union-type` (rather than general principles). So I'm sometimes not sure what the most idiomatic way to use it is, or if a feature I'd like to see would be an appropriate addition.
